src/main/java/com/aws/greengrass/componentmanager/ComponentServiceHelper.java

```diff
@@ -84,14 +85,25 @@ ResolvedComponentVersion resolveComponentVersion(String componentName, Semver lo
                         .maxRetryInterval(retryInterval).maxAttempt(CLIENT_RETRY_COUNT)
                         .retryableExceptions(Arrays.asList(DeviceConfigurationException.class)).build();
 
-        try (GreengrassV2DataClient greengrasV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
+        try (GreengrassV2DataClient greengrassV2DataClient = RetryUtils.runWithRetry(clientExceptionRetryConfig,
                 clientFactory::fetchGreengrassV2DataClient, "get-greengrass-v2-data-client", logger)) {
-            result = greengrasV2DataClient.resolveComponentCandidates(request);
+            result = greengrassV2DataClient.resolveComponentCandidates(request);
         } catch (ResourceNotFoundException e) {
-            logger.atDebug().kv("componentName", componentName).kv("versionRequirements", versionRequirements)
-                    .log("No applicable version found in cloud registry", e);
-            throw new NoAvailableComponentVersionException("No cloud component version satisfies the requirements.",
-                    componentName, versionRequirements);
+            if (e.getMessage().contains("claim platform")) {
+                logger.atInfo().kv("componentName", componentName).kv("versionRequirements", versionRequirements)
+                        .log("The platform claimed by the component is incompatible", e);
+                throw new IncompatiblePlatformClaimByComponentException("The platform claimed by the component is incompatible.",
+                        componentName, platform.name());
```

Comment:
not sure what the platform name even is, we never set that field at all. The platform is a map of attributes, that's what you need to pass here.

Reply:
`platformResolver.getCurrentPlatform()`

Reply:
I got the "name" member from ComponentPlatform class of the Greengrassv2 data plane library, but that's incorrect

Reply:
How do you plan to test this change? You can either do an integ test and mock the exception returned by cloud. Or you will need to add a UAT.